src/custom-elements/src/components/progress/progress.tsx

```diff
@@ -0,0 +1,52 @@
+import { Component, Prop, Watch } from '@stencil/core';
+
+@Component({
+  tag: 'chi-progress',
+  styleUrl: 'progress.scss',
+  scoped: true
+})
+export class Progress {
+  /**
+   *  to set the value of a progress bar.
+   */
+  @Prop({ reflectToAttr: true }) value: string;
+
+  /**
+   *  to set the min value of a progress bar.
+   */
+  @Prop({ reflectToAttr: true }) min = '0';
+
+  /**
+   *  to set the max value of a progress bar.
+   */
+  @Prop({ reflectToAttr: true }) max = '100';
+
+  /**
+   *  to indicate a progress bar state { info, warning, danger, success }.
+   */
+  @Prop({ reflectToAttr: true }) state = 'success';
+
+  @Watch('state')
+  stateValidation(newValue: string) {
+    if (newValue && !['info', 'warning', 'danger', 'success'].includes(newValue)) {
```

Comment:
What about turning this list into a const?

Reply:
@srinath-koneru yes, please, add this list as a const in the const folder.

Reply:
@dani-cl-madrid @SergioQCL roger that. Do you have any variable name in mind so that it can be re-used in any other component?

Reply:
@srinath-koneru I think it could be necessary to create more than one of those lists per component so you could name them {ce-name}_{prop} --> progress_states or similar.

Reply:
@srinath-koneru @SergioQCL No, this list can be shared. It is a well defined list in chi foundations:
https://assets.ctl.io/chi/1.2.2/foundations/color/#semantic
Include it in the colors.ts file and you can use SEMANTIC_COLORS as constant name.